app/controllers/projects/samples_controller.rb

```diff
@@ -32,11 +32,24 @@ def show
       @tab = params[:tab]
       @table_listing = if @tab == 'metadata'
                          @sample.metadata_with_provenance
+                       elsif @tab == 'history'
+                         @log_data = @sample.log_data_without_changes
                        else
                          @sample.attachments
                        end
     end
 
+    def view_history_version
+      authorize! @sample.project, to: :view_history?
+
+      @log_data = @sample.log_data_with_changes(params[:version])
+      respond_to do |format|
+        format.turbo_stream do
```

Comment:
**question:** Do we need an error state if something goes wrong? perhaps just and alert field that says `History is not available at this time?`

Reply:
Going to do this in another PR as it will be used by the other history views as well.

Reply:
Sounds good, thanks